assemble/bin/accumulo

```diff
@@ -15,173 +15,189 @@
-   exit 1
-fi
-if [ -z "${ZOOKEEPER_HOME}" -o ! -d "${ZOOKEEPER_HOME}" ]; then
-   echo "ZOOKEEPER_HOME is not set or is not a directory.  Please make sure it's set globally or in conf/accumulo-env.sh"
-   exit 1
-fi
-
-# This is default for hadoop 2.x;
-#   for another distribution, specify (DY)LD_LIBRARY_PATH
-#   explicitly in ${ACCUMULO_HOME}/conf/accumulo-env.sh
-#   usually something like:
-#     ${HADOOP_PREFIX}/lib/native/${PLATFORM}
-if [ -e "${HADOOP_PREFIX}/lib/native/libhadoop.so" ]; then
-   LIB_PATH="${HADOOP_PREFIX}/lib/native"
-   LD_LIBRARY_PATH="${LIB_PATH}:${LD_LIBRARY_PATH}"     # For Linux
-   DYLD_LIBRARY_PATH="${LIB_PATH}:${DYLD_LIBRARY_PATH}" # For Mac
-fi
-
-# Export the variables just in case they are not exported
-# This makes them available to java
-export JAVA_HOME HADOOP_PREFIX ZOOKEEPER_HOME LD_LIBRARY_PATH DYLD_LIBRARY_PATH
-
-# Strip the instance from $1
-APP=$1
-# Avoid setting an instance unless it's necessary to ensure consistency in filenames
-INSTANCE=""
-# Avoid setting a pointless system property
-INSTANCE_OPT=""
-if [[ "$1" =~ ^tserver-[1-9][0-9]*$ ]]; then
-  APP="$(echo "$1" | cut -d'-' -f1)"
-  # Appending the trailing underscore to make single-tserver deploys look how they did
-  INSTANCE="$(echo "$1" | cut -d'-' -f2)_"
-
-  #Rewrite the input arguments
-  set -- "$APP" "${@:2}"
-
-  # The extra system property we'll pass to the java cmd
-  INSTANCE_OPT="-Daccumulo.service.instance=${INSTANCE}"
-fi
+  echo "Successfully installed native library"
+}
 
-#
-# app isn't used anywhere, but it makes the process easier to spot when ps/top/snmp truncate the command line
-JAVA="${JAVA_HOME}/bin/java"
-exec "$JAVA" "-Dapp=$1" \
-   $INSTANCE_OPT \
-   $ACCUMULO_OPTS \
-   -classpath "${CLASSPATH}" \
-   -XX:OnOutOfMemoryError="${ACCUMULO_KILL_CMD:-kill -9 %p}" \
-   -XX:-OmitStackTraceInFastThrow \
-   -Djavax.xml.parsers.DocumentBuilderFactory=com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl \
-   -Dorg.apache.accumulo.core.home.dir="${ACCUMULO_HOME}" \
-   -Dhadoop.home.dir="${HADOOP_PREFIX}" \
-   -Dzookeeper.home.dir="${ZOOKEEPER_HOME}" \
-   org.apache.accumulo.start.Main \
-   "$@"
+function main() {
+
+  # Start: Resolve Script Directory
+  SOURCE="${BASH_SOURCE[0]}"
+  while [ -h "${SOURCE}" ]; do # resolve $SOURCE until the file is no longer a symlink
+     bin="$( cd -P "$( dirname "${SOURCE}" )" && pwd )"
+     SOURCE="$(readlink "${SOURCE}")"
+     [[ "${SOURCE}" != /* ]] && SOURCE="${bin}/${SOURCE}" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+  done
+  bin="$( cd -P "$( dirname "${SOURCE}" )" && pwd )"
+  basedir=$( cd -P "${bin}"/.. && pwd )
+  # Stop: Resolve Script Directory
+
+  export ACCUMULO_CONF_DIR="${ACCUMULO_CONF_DIR:-$basedir/conf}"
+  export ACCUMULO_LIB_DIR="${ACCUMULO_LIB_DIR:-$basedir/lib}"
+
+  if [[ "$1" == "create-config" ]]; then
+    "${ACCUMULO_LIB_DIR}/scripts/config.sh" ${*:2}
```

Comment:
If possible, we should see if we can't roll up config.sh into this script or push it down into accumulo-env.sh
The reasoning is that it seems to just being doing environment prep for the scripts... and that can be done inside the scripts themselves or by the user, based on their own environment.
Scripts should have environment prerequisites, and user environment script should satisfy those prerequisites. There doesn't need to be a middle-man script.